substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -94,17 +94,246 @@ private PosixJavaNetSubstitutions() {
+        //         * point.
+        //         */
+        //            if (packetBufferLen > MAX_PACKET_LEN) {
+        //                packetBufferLen = MAX_PACKET_LEN;
+        //            }
+        //            fullPacket = (char *)malloc(packetBufferLen);
+        CCharPointer fullPacket = LibC.malloc(WordFactory.unsigned(packetBufferLen));
+        //
+        //            if (!fullPacket) {
+        //                JNU_ThrowOutOfMemoryError(env, "Send buffer native heap allocation failed");
+        //                return;
+        //            } else {
+        //                mallocedPacket = JNI_TRUE;
+        //            }
+        //        } else {
+        //            fullPacket = &(BUF[0]);
+        //        }
+        //
+        //        (*env)->GetByteArrayRegion(env, packetBuffer, packetBufferOffset, packetBufferLen,
+        //                                   (jbyte *)fullPacket);
+        //#ifdef AF_INET6
+        //        if (trafficClass != 0 && ipv6_available()) {
+        //            NET_SetTrafficClass((struct sockaddr *)&rmtaddr, trafficClass);
+        //        }
+        //#endif /* AF_INET6 */
+        //
+        //
+        //    /*
+        //     * Send the datagram.
+        //     *
+        //     * If we are connected it's possible that sendto will return
+        //     * ECONNREFUSED indicating that an ICMP port unreachable has
+        //     * received.
+        //     */
+        //                ret = NET_SendTo(fd, fullPacket, packetBufferLen, 0,
+        //                                 (struct sockaddr *)rmtaddrP, len);
+        //CCharPointer bufP = WordFactory.nullPointer();
+        VmPrimsJNI.GetByteArrayRegion(packet.getData(), 0, packetBufferLen, fullPacket);
+        int ret = JavaNetNetUtilMD.NET_SendTo(fd, fullPacket, packetBufferLen, 0, rmtaddr, JavaNetNetUtilMD.SOCKADDR_LEN());
+        LibC.free(fullPacket);
+        //
+        //        if (ret < 0) {
+        //            switch (ret) {
+        //                case JVM_IO_ERR :
+        //                    if (errno == ECONNREFUSED) {
+        //                        JNU_ThrowByName(env, JNU_JAVANETPKG "PortUnreachableException",
+        //                                        "ICMP Port Unreachable");
+        //                    } else {
+        //                        NET_ThrowByNameWithLastError(env, "java/io/IOException", "sendto failed");
+        //                    }
+        //                    break;
+        //
+        //                case JVM_IO_INTR:
+        //                    JNU_ThrowByName(env, "java/io/InterruptedIOException",
+        //                                    "operation interrupted");
+        //                    break;
+        //            }
+        //        }
+        //
+        //        if (mallocedPacket) {
+        //            free(fullPacket);
+        //        }
+        //        return;
```

Comment:
What happened to all this error checking?

(Ah, see, here is a case where the C code "throws" and exception and then runs some more code (`free(fullPacket)`).  I would translate that as a `try { .... } finally { .... }`.